### ISSUE TITLE
v1.8.7 linker command failed: undefined reference to `fc::optional<double>::~optional()'

### ISSUE THREAD

**Issue by theblockstalk** [EOSIO/eos#8412]
I am getting this issue using the autobuild script on Ubuntu 18.0.4 with the eosio v1.8.7 on master branch

```git clone https://github.com/EOSIO/eos --recursive
cd eos
./scripts/eosio_build.sh -o Debug -m
```

(the same issue is reported in https://github.com/EOSIO/eos/issues/8216)
I have updated to clang 8 but still get the same issue:

```
[100%] Linking CXX executable nodeos
cd /home/developers/Documents/Apps/eosio/eos/build/programs/nodeos && /usr/bin/cmake -E cmake_link_script CMakeFiles/nodeos.dir/link.txt --verbose=false
../../plugins/producer_plugin/libproducer_plugin.a(producer_plugin.cpp.o): In function `eosio::producer_plugin::get_runtime_options() const':
/home/developers/Documents/Apps/eosio/eos/plugins/producer_plugin/producer_plugin.cpp:959: undefined reference to `fc::optional<double>::~optional()'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
programs/nodeos/CMakeFiles/nodeos.dir/build.make:184: recipe for target 'programs/nodeos/nodeos' failed
make[2]: *** [programs/nodeos/nodeos] Error 1
make[2]: Leaving directory '/home/developers/Documents/Apps/eosio/eos/build'
CMakeFiles/Makefile2:4096: recipe for target 'programs/nodeos/CMakeFiles/nodeos.dir/all' failed
make[1]: *** [programs/nodeos/CMakeFiles/nodeos.dir/all] Error 2
make[1]: Leaving directory '/home/developers/Documents/Apps/eosio/eos/build'
Makefile:140: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by mulspace** [comment]
It is a know issue of clang7.
Please upgrade to clang8.

**Comment by theblockstalk** [comment]
I already tried upgrading to clang 8 as per https://github.com/EOSIO/eos/issues/8216 but without success.

Did you update clang before or after initially installing? And how have you upgraded clang?

**Comment by spoonincode** [comment]
Fixed in 1.8.9 & 2.0 rc
